src/test/java/com/codeus/winter/util/PropertiesResourceReaderTest.java

```diff
@@ -0,0 +1,75 @@
+package com.codeus.winter.util;
+
+import com.codeus.winter.exception.PropertySourceException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.List;
+import java.util.Properties;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+class PropertiesResourceReaderTest {
```

Comment:
How about an idea of having one abstract `ResourceReaderTest` where all tests are defined and one abstract `setUp()` method. Then you provide two implementations `ResourceReaderTest` for `ResourceReader` implementations.

By doing this way we reduce code duplication.